src/sonic_ax_impl/mibs/ietf/rfc3433.py

```diff
@@ -303,19 +429,185 @@ def update_data(self):
+                        self.ent_phy_sensor_value_map[sub_id], \
+                        self.ent_phy_sensor_oper_state_map[sub_id] = mib_values
 
                     self.sub_ids.append(sub_id)
 
+    def update_psu_sensor_data(self):
+        if not self.psu_sensor:
+            return
+
+        for psu_sensor_entry in self.psu_sensor:
+            psu_name = psu_sensor_entry.split(mibs.TABLE_NAME_SEPARATOR_VBAR)[-1]
+            psu_relation_info = Namespace.dbs_get_all(self.statedb, mibs.STATE_DB,
+                                                      mibs.physical_entity_info_table(psu_name))
+            psu_position, psu_parent_name = get_db_data(psu_relation_info, PhysicalRelationInfoDB)
+            if is_null_empty_str(psu_position):
+                continue
+            psu_position = int(psu_position)
+            psu_sub_id = get_psu_sub_id(psu_position)
+
+            psu_sensor_entry_data = Namespace.dbs_get_all(self.statedb, mibs.STATE_DB, psu_sensor_entry)
+
+            if not psu_sensor_entry_data:
+                continue
+
+            sensor_data_list = PSUSensorData.create_sensor_data(psu_sensor_entry_data)
+            for sensor_data in sensor_data_list:
+                raw_sensor_value = sensor_data.get_raw_value()
+                if is_null_empty_str(raw_sensor_value):
+                    continue
+                sensor = sensor_data.get_sensor_interface()
+                sub_id = get_psu_sensor_sub_id(psu_sub_id, sensor_data.get_name().lower())
+
+                try:
+                    mib_values = sensor.mib_values(raw_sensor_value)
+                except (ValueError, ArithmeticError):
+                    mibs.logger.error("Exception occurred when converting"
+                                      "value for sensor {} PSU {}".format(sensor, psu_name))
+                    continue
+                else:
+                    self.ent_phy_sensor_type_map[sub_id], \
+                        self.ent_phy_sensor_scale_map[sub_id], \
+                        self.ent_phy_sensor_precision_map[sub_id], \
+                        self.ent_phy_sensor_value_map[sub_id], \
+                        self.ent_phy_sensor_oper_state_map[sub_id] = mib_values
+
+                    self.sub_ids.append(sub_id)
+
+    def update_fan_sensor_data(self):
+        if not self.fan_sensor:
+            return
+
+        fan_parent_sub_id = 0
+        for fan_sensor_entry in self.fan_sensor:
+            fan_name = fan_sensor_entry.split(mibs.TABLE_NAME_SEPARATOR_VBAR)[-1]
+            fan_relation_info = Namespace.dbs_get_all(self.statedb, mibs.STATE_DB,
+                                                      mibs.physical_entity_info_table(fan_name))
+
+            fan_position, fan_parent_name = get_db_data(fan_relation_info, PhysicalRelationInfoDB)
+            if is_null_empty_str(fan_position):
+                continue
+
+            fan_position = int(fan_position)
+
+            if "chassis" in fan_parent_name:
```

Comment:
"chassis" and "PSU" can be defined as string constants. "chassis" is used in another location here.

Reply:
fixed